File: tcpclient/src/main/java/com/pointr/tcp/java/sparcle/AppContainerServiceDriver.java
package com.pointr.tcp.java.sparcle;

import com.pointr.tcp.rpc.ServerFactory;
import com.pointr.tcp.rpc.TcpClient;
import com.pointr.tcp.rpc.TcpServer;
import com.pointr.tcp.util.TcpUtils;
import scala.collection.immutable.Iterable;

public class AppContainerServiceDriver {

  public static String run() throws InterruptedException {
    System.setProperty("logger.level","3");
    String serverConfPath = "src/main/resources/appContainer-server.java.yaml";
    String  confPath = "src/main/resources/appContainer.java.yaml";
    Iterable<TcpServer> servers = ServerFactory.create(serverConfPath);
    Thread.sleep(200);
    TcpClient client = TcpClient.createClientFromArgs(
        new String[]{TcpUtils.getLocalHostname(),"" + TcpServer.DefaultPort(),confPath});
    String out = (String)client.serviceIf().run();
    System.out.println(out);
    return out;
  }

  public static void main(String[] args) throws InterruptedException {
    run();
  }

}

File: tcpclient/src/main/scala/com/pointr/tcp/sparcle/AppContainerServiceDriver.scala
package com.pointr.tcp.sparcle

import com.pointr.tcp.rpc.{ServerFactory, ServiceIf, TcpClient, TcpServer}
import com.pointr.tcp.util.{Logger, TcpUtils}

object AppContainerServiceDriver {

  def run(args: Array[String]) = {
    System.setProperty("logger.level","3")
    val serverConfPath = "src/main/resources/appContainer-server.yaml"
    val confPath = "src/main/resources/appContainer.yaml"
    val servers = ServerFactory.create(serverConfPath)
    Thread.sleep(200)
    val client = TcpClient.createClientFromArgs(Array(TcpUtils.getLocalHostname,"" + TcpServer.DefaultPort,confPath))
    val out = client.serviceIf.run().asInstanceOf[String]
    println(out)
    out
  }

  def main(args: Array[String]): Unit = {
    run(args)
  }

}
